File: q4.c
/*  a simple program that takes the average of an integer list */
#include <stdio.h>

int main() {
    int a[50],n,i,j,avg=0;
    printf("Enter the number of elements in the list\n");
    scanf("%d",&n);
    printf("Enter the elements \n");
    for(i=0;i<n;i++){
        scanf("%d",&a[i]);
    }
    printf("the elemnts in the list are \n");
    for(i=0;i<n;i++){
       printf ("%d\t",a[i]);
        avg=avg+a[i];
    }
    avg=avg/n;
    printf("\nthe average of the numbers is %d",avg);
    
    return 0;
};

File: q4.py
#Calculating the average of an integer list:

n = int(input("Enter the number of elements in the list: "))
elements = []
for i in range(n):
    num=int(input("enter number "))
    elements.append(num)
total = sum(elements)
average = total / n if n != 0 else 0
print("The average of the numbers is", average)
